src/extensions/default/NavigationAndHistory/NavigationProvider.js

```diff
@@ -0,0 +1,359 @@
+        Commands                = brackets.getModule("command/Commands"),
+        Menus                   = brackets.getModule("command/Menus"),
+        KeyBindingManager       = brackets.getModule("command/KeyBindingManager"),
+        FileSystem              = brackets.getModule("filesystem/FileSystem");
+
+    var KeyboardPrefs = JSON.parse(require("text!keyboard.json"));
+    
+    // Command constants for navigation history
+    var NAVIGATION_JUMP_BACK      = "navigation.jump.back",
+        NAVIGATION_JUMP_FWD       = "navigation.jump.fwd";
+    
+    // The latency time to capture an explicit cursor movement as a navigation frame
+    var NAV_FRAME_CAPTURE_LATENCY = 2000;
+    
+    /*
+    * Contains list of most recently known cursor positions.
+    * @private
+    * @type {Array.<Object>}
+    */
+    var jumpToPosStack = [];
+    
+    /*
+    * Contains list of most recently traversed cursor positions using NAVIGATION_JUMP_BACK command.
+    * @private
+    * @type {Array.<Object>}
+    */
+    var jumpedPosStack = [],
+        captureTimer,
+        activePosNotSynched = false,
+        jumpInProgress,
+        command_JumpBack,
+        command_JumpFwd,
+        cmMarkers = {};
+     
+   /**
+    * Function to enable/disable navigation command based on cursor positions availability.
+    * @private
+    */
+    function _validateNavigationCmds() {
+        if (jumpToPosStack.length > 0) {
+            command_JumpBack.setEnabled(true);
+        } else {
+            command_JumpBack.setEnabled(false);
+        }
+        
+        if (jumpedPosStack.length > 0) {
+            command_JumpFwd.setEnabled(true);
+        } else {
+            command_JumpFwd.setEnabled(false);
+        }
+    }
+    
+    /**
+    * Function to check existence of a file entry
+    * @private
+    */
+    function _checkIfExist(entry) {
+        var deferred = new $.Deferred(),
+            fileEntry = FileSystem.getFileForPath(entry.file);
+
+        if (entry.inMem) {
+            var indxInWS = MainViewManager.findInWorkingSet(entry.paneId, entry.file);
+            // Remove entry if InMemoryFile is not found in Working set
+            if (indxInWS === -1) {
+                deferred.reject();
+            } else {
+                deferred.resolve();
+            }
+        } else {
+            fileEntry.exists(function (err, exists) {
+                if (!err && exists) {
+                    deferred.resolve();
+                } else {
+                    deferred.reject();
+                }
+		    });
+	    }
+
+        return deferred.promise();
+    }
+        
+   /**
+    * Prototype to capture a navigation frame and it's various data/functional attributues
+    */
+    function NavigationFrame(editor, selectionObj) {
+        this.cm = editor._codeMirror;
+        this.file = editor.document.file._path;
+        this.inMem = editor.document.file.constructor.name === "InMemoryFile" ? true : false;
```

Comment:
`this.inMem = editor.document.file.constructor.name === "InMemoryFile"`

is sufficent.